lib/inspec/resources/oracledb_session.rb

```diff
@@ -87,6 +80,29 @@ def pre_flight_check
       return fail_resource "Can't run Oracle checks without authentication" unless su_user && (user || password)
       return fail_resource "You must provide a service name for the session" unless service
     end
+
+    # 3 commands
+    # regular user password
+    # using a db_role
+    # su, using a db_role
+    def command_builder(format_options, query)
+      verified_query = verify_query(query)
+      sql_prefix, sql_postfix = ""
+      if inspec.os.windows?
+        sql_prefix = %{@'\n#{format_options}\n#{verified_query}\nEXIT\n'@ | }
+      else
+        sql_postfix = %{ <<'EOC'\n#{format_options}\n#{verified_query}\nEXIT\nEOC}
+      end
+
+      if @db_role.nil?
+        %{#{sql_prefix}#{bin} "#{user}"/"#{password}"@#{host}:#{port}/#{@service}#{sql_postfix}}
+      elsif @su_user.nil?
+        %{#{sql_prefix}#{bin} "#{user}"/"#{password}"@#{host}:#{port}/#{@service} as #{@db_role}#{sql_postfix}}
```

Comment:
I can't help but think that there are more opportunities to clean this up. Something like

```ruby
sql_postfix = " as #{@db_role}#{sql_postfix}" if @db_role
```

and then remove the first `if` entirely.